backend/accounts.go

```diff
@@ -228,19 +228,36 @@ func (backend *Backend) LookupInsuredAccounts(accountCode accountsTypes.Code) ([
 	// if any account insurance status changed, persist the change and reinitialize the accounts.
 	statusChange := false
 	err = backend.config.ModifyAccountsConfig(func(accountsConfig *config.AccountsConfig) error {
-		for code, insured := range bitsuranceAccounts {
-			accountConfig := accountsConfig.Lookup(code)
+		for _, bitsuranceAccount := range bitsuranceAccounts {
+			bitsuranceStatus := string(bitsuranceAccount.Status)
+			accountConfig := accountsConfig.Lookup(bitsuranceAccount.AccountCode)
 			if accountConfig == nil {
-				return errp.Newf("Could not find account %s", code)
+				return errp.Newf("Could not find account %s", bitsuranceAccount.AccountCode)
 			}
-			if accountConfig.Insured != insured {
-				accountConfig.Insured = insured
-				backend.log.Infof("Account [%s] insurance status changed to %v", code, insured)
+			if accountConfig.InsuranceStatus != bitsuranceStatus {
+				backend.log.Infof("Account [%s] insurance status changed to %v", bitsuranceAccount.AccountCode, bitsuranceStatus)
+				canceled := bitsuranceStatus == string(bitsurance.CanceledStatus) || bitsuranceStatus == string(bitsurance.InactiveStatus)
+				if canceled {
```

Comment:
just an idea: couldn't you return a flag from this function in this case, and in the handler return that result to the frontend, and then let the frontend react to it by doing a notification?

to avoid having this frontend stuff here deep in the backend.

or is this function also called in other places independently from the handler that still needs to notify the user?

Reply:
another idea:

send an event to the frontend and have the frontend react to that? `backend.Notify(observable.Event{...})`

Reply:
The function is called both on the dashboard and on the account page, but the notification is only displayed on the account page. This solution allows notifying the user even if the app is closed before the alert is actually displayed. As an alternative, we could subscribe to the event and persist the config from the frontend, but I'm not sure that it is going to simplify much the code.